keyboards/lily58/keymaps/nazu/keymap.c

```diff
@@ -0,0 +1,269 @@
+//   {     23,     22,     21,     20,     19,     18,                     17,     16,     15,     14,     13,     12},
+//   {     35,     34,     33,     32,     31,     30,                     29,     28,     27,     26,     25,     24},
+//   {     49,     48,     47,     46,     45,     44,     43,     42,     41,     40,     39,     38,     37,     36},
+//   {                             57,     56,     55,     54,     53,     52,     51,     50                        },
+
+    {  0,  1,  2,  3,  4,  5 },
+    {  6,  7,  8,  9, 10, 11 },
+    { 12, 13, 14, 15, 16, 17 },
+    { 18, 19, 20, 21, 22, 23 },
+    { NO_LED, 24, 25, 26, 27, 28 },
+    { 29, 30, 31, 32, 33, 34 },
+    { 35, 36, 37, 38, 39, 40 },
+    { 41, 42, 43, 44, 45, 46 },
+    { 47, 48, 49, 50, 51, 52 },
+    { NO_LED, 53, 54, 55, 56, 57 },
+}, {
+    // LED Index to Physical Position
+    {   0,   0 }, {  16,   0 }, {  32,   0 }, {  48,   0 }, {  64,   0 }, {  80,   0 },
+    {   0,  16 }, {  16,  16 }, {  32,  16 }, {  48,  16 }, {  64,  16 }, {  80,  16 },
+    {   0,  32 }, {  16,  32 }, {  32,  32 }, {  48,  32 }, {  64,  32 }, {  80,  32 },
+    {   0,  48 }, {  16,  48 }, {  32,  48 }, {  48,  48 }, {  64,  48 }, {  80,  48 },
+                  {  48,  64 }, {  64,  64 }, {  80,  64 }, {  96,  64 }, {  96,  48 },
+    { 128,   0 }, { 144,   0 }, { 160,   0 }, { 176,   0 }, { 192,   0 }, { 224,   0 },
+    { 128,  16 }, { 144,  16 }, { 160,  16 }, { 176,  16 }, { 192,  16 }, { 224,  16 },
+    { 128,  32 }, { 144,  32 }, { 160,  32 }, { 176,  32 }, { 192,  32 }, { 224,  32 },
+    { 128,  48 }, { 144,  48 }, { 160,  48 }, { 176,  48 }, { 192,  48 }, { 224,  48 },
+                  { 128,  64 }, { 144,  64 }, { 160,  64 }, { 112,  64 }, { 112,  48 },
+}, {
+    // LED Index to Flag (1 - modifier, 2 - underglow, 4 - key backlight)
+    1, 4, 4, 4, 4, 4,
+    1, 4, 4, 4, 4, 4,
+    1, 4, 4, 4, 4, 4,
+    1, 4, 4, 4, 4, 4,
+       1, 1, 1, 1, 1,
+    1, 4, 4, 4, 4, 4,
+    1, 4, 4, 4, 4, 4,
+    1, 4, 4, 4, 4, 4,
+    1, 4, 4, 4, 4, 4,
+       1, 1, 1, 1, 1,
+} };
+
+#endif
+
+/* Rotary encoder related code */
+#ifdef ENCODER_ENABLE
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) { // Encoder on master side
+    if(IS_LAYER_ON(_RAISE)) { // on Raise layer
+      // Cursor control
+      if (clockwise) {
+          tap_code(KC_MNXT);
+      } else {
+          tap_code(KC_MPRV);
+      }
+    }
+    else {
+      if (clockwise) {
+          tap_code(KC_VOLU);
+      } else {
+          tap_code(KC_VOLD);
+      }
+    }
+  }
+  else if (index == 1) { // Encoder on slave side
+    if(IS_LAYER_ON(_LOWER)) { // on Lower layer
+      //
+      if (clockwise) {
+          tap_code(KC_RIGHT);
+      } else {
+          tap_code(KC_LEFT);
+      }
+    }
+    else {
+      if (clockwise) {
+          tap_code(KC_DOWN);
+      } else {
+          tap_code(KC_UP);
+      }
+    }
+  }
+    return true;
+}
+#endif // ENCODER_ENABLE
```

Comment:
This entire block can be replaced with the encoder map feature. 

https://docs.qmk.fm/#/feature_encoders?id=encoder-map